### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by Skiryy** [28Pollux28/ImageWorldGenerator#20]
**Describe the bug**
Some biomes have patterned lava/ice pockets generating in the ocean next to them 

**Expected behavior**
I expected there just to be normal ocean

**Screenshots**
![image](https://user-images.githubusercontent.com/84138927/120108218-7fcd8d80-c164-11eb-85d0-cb3526a4f134.png)

**Version**
Mod Version :  1.16.5
Environment : Server

**Comment by 28Pollux28** [comment]
Could you link down the image or part of the image you used to generate the world where this happened ?

**Comment by Skiryy** [comment]
![image](https://user-images.githubusercontent.com/84138927/120110072-7ba56e00-c16c-11eb-9a14-2adc52c507a1.png)

**Comment by 28Pollux28** [comment]
Would you mind sending the full image, the imggen.json config file, the coordinates where this happened and the world seed?

**Comment by Skiryy** [comment]
For sure, the 
![BiomeImage](https://user-images.githubusercontent.com/84138927/120114104-dabfae80-c17d-11eb-86b6-ddd8746415bd.png)
"{
  "configVersion": "1.1.3",
  "biomesImageName": "biomeImage.png",
  "biomesImageScale": 2.0,
  "repeatBiomeImage": false,
  "repeatMirrorBiomeImage": false,
  "defaultBiome": "minecraft:ocean",
  "continuousGen": true,
  "largeBiomes": false,
  "customBiomes": [
    {
      "biomeID": "modid:biomeid",
      "RGB": "0x000000"
    },
    {
      "biomeID": "modid:biomeid",
      "RGB": "0xFFFFFF"
    }
  ],
  "customHeightMap": false,
  "heightMapName": "HeightMap.png",
  "heightMapScale": 1.0,
  "repeatHeightMapImage": false,
  "repeatMirrorHeightMapImage": false,
  "seaLevel": 63,
  "placeVanillaStructures": true,
  "customStructures": false,
  "customStructuresImage": "structureMap.png",
  "repeatStructureImage": false,
  "repeatMirrorStructureImage": false,
  "useBlackList": false,
  "blackListApplication": 2,
  "structureBlacklist": [
    "modid:structureid",
    "modid:structureid"
  ],
  "customStructuresRGB": [
    {
      "structureID": "modid:structureid",
      "RGB": "0x123456"
    },
    {
      "structureID": "modid:structureid",
      "RGB": "0x024680"
    }
  ],
  "lakeFormationPercentChance": 0.5
}"

and around -6700, 2040
seed: 7222263084176951517

Also if this is of any importance: I pregenerated the terrain with a chunk generator on my fabric server and then transferred it to a spigot server.

**Comment by 28Pollux28** [comment]
Ok thanks, I'll try to have a look during the week, will keep you posted.

**Comment by 28Pollux28** [comment]
Hi ! I just had the time to look over your issue. So the first thing I noticed was that the biome where those lava column generated was mainly basalt_deltas. After some investigation, I found that your color gradients between beach and deep ocean was the problem, as some values used to make the gradient come too close to the Basalt Deltas biome color.
For example, if you look closely, pixel (1662,5986), hex code is 0x393648, which is very close to the 0x403636 of the Basalt Deltas color.
What I suggest you to do : don't use gradients on biome borders. Use sharp borders that way there won't be issues like this one.

Pollux
